pkg/apiserver/rest/usecase/envbinding.go

```diff
@@ -0,0 +1,193 @@
+		AppPrimaryKey: app.PrimaryKey(),
+		Name:          envName,
+	}
+	err := c.ds.Get(ctx, &envBinding)
+	if err != nil {
+		return nil, err
+	}
+	return &envBinding, nil
+}
+
+func (c *envBindingUsecaseImpl) UpdateEnvBinding(ctx context.Context, app *model.Application,
+	envName string, envUpdate apisv1.PutApplicationEnvRequest) (*apisv1.EnvBinding, error) {
+	envBinding, err := c.getBindingByEnv(ctx, app, envName)
+	if err != nil {
+		return nil, err
+	}
+	var envBind model.EnvBinding
+	// update env-binding base
+	for i, env := range app.EnvBinding {
+		if env.Name == envName {
+			if envUpdate.Description != nil {
+				app.EnvBinding[i].Description = *envUpdate.Description
+			}
+			if envUpdate.Alias != nil {
+				app.EnvBinding[i].Alias = *envUpdate.Alias
+			}
+			if envUpdate.TargetNames != nil {
+				app.EnvBinding[i].TargetNames = *envUpdate.TargetNames
+			}
+			if envUpdate.ComponentSelector == nil {
+				app.EnvBinding[i].ComponentSelector = nil
+			} else {
+				app.EnvBinding[i].ComponentSelector = &model.ComponentSelector{
+					Components: envUpdate.ComponentSelector.Components,
+				}
+			}
+			envBind = *app.EnvBinding[i]
+		}
+	}
+	// update envbinding
+	if err := c.ds.Put(ctx, &envBind); err != nil {
+		return nil, err
+	}
+	re := &apisv1.EnvBinding{
+		Name:        envBinding.Name,
+		Alias:       envBinding.Alias,
+		Description: envBinding.Description,
+		TargetNames: envBinding.TargetNames,
+	}
+	if envBinding.ComponentSelector != nil {
+		re.ComponentSelector = (*apisv1.ComponentSelector)(envBinding.ComponentSelector)
+	}
+	return re, nil
+}
+
+func (c *envBindingUsecaseImpl) DeleteEnvBinding(ctx context.Context, app *model.Application, envName string) error {
+	envBindings, err := c.GetEnvBindings(ctx, app)
+	if err != nil {
+		return err
+	}
+	for _, envBinding := range envBindings {
+		if err := c.ds.Delete(ctx, &model.EnvBinding{AppPrimaryKey: app.PrimaryKey(), Name: envBinding.Name}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *envBindingUsecaseImpl) DetailEnvBinding(ctx context.Context, envBinding *model.EnvBinding) (*apisv1.DetailEnvBindingResponse, error) {
+	return &apisv1.DetailEnvBindingResponse{
+		EnvBindingBase: *converEnvbindingModelToBase(envBinding),
+	}, nil
+}
+
+func convertCreateReqToEnvBindingModel(app *model.Application, req apisv1.CreateApplicationEnvRequest) model.EnvBinding {
+	envBinding := model.EnvBinding{
+		AppPrimaryKey:     app.Name,
+		Name:              req.Name,
+		Alias:             req.Alias,
+		Description:       req.Description,
+		TargetNames:       req.TargetNames,
+		ComponentSelector: (*model.ComponentSelector)(req.ComponentSelector),
+	}
+	return envBinding
+}
+
+func converEnvbindingModelToBase(envBinding *model.EnvBinding) *apisv1.EnvBindingBase {
+	ebb := &apisv1.EnvBindingBase{
+		Name:              envBinding.Name,
+		Alias:             envBinding.Alias,
+		Description:       envBinding.Description,
+		TargetNames:       envBinding.TargetNames,
```

Comment:
Here you can load the alias and description of DeliveryTarget.

Reply:
The relationship between envbinding and deliveryTarget is one-to-many

Reply:
The purpose of loading the alias is to serve the UI.
suggest:
```
targets:
  - name: xx
     alias: xxxx
```